gusto/timeloop.py

```diff
@@ -228,16 +228,19 @@ def set_reference_profiles(self, reference_profiles):
             else:
                 raise ValueError(f'When initialising reference profile {field_name}'
                                  + ' the passed profile must be a Function')
+            # if field name is not prognostic we need to add it
             ref.interpolate(profile)
-
             # Assign profile to X_ref belonging to equation
             if isinstance(self.equation, PrognosticEquationSet):
-                assert field_name in self.equation.field_names, \
-                    f'Cannot set reference profile as field {field_name} not found'
-                idx = self.equation.field_names.index(field_name)
-                X_ref = self.equation.X_ref.subfunctions[idx]
-                X_ref.assign(ref)
-
+                if field_name in self.equation.field_names:
+                    idx = self.equation.field_names.index(field_name)
+                    X_ref = self.equation.X_ref.subfunctions[idx]
+                    X_ref.assign(ref)
+                else:
+                    # reference profile of a diagnostic
+                    # warn user in case they made a typo
+                    logger.warning(f'Setting reference profile for diagnostic {field_name}')
+                    # dont need to do anythin else as value in field container has already been set
```

Comment:
Could you fix the typos in this comment?
```suggestion
                    # Don't need to do anything else as value in field container has already been set
```